corehq/motech/dhis2/templates/dhis2/dataset_map.html

```diff
@@ -9,8 +9,28 @@
 
 {% initial_page_data 'dataset_maps' dataset_maps %}
 {% initial_page_data 'send_data_url' send_data_url %}
+{% initial_page_data 'is_json_ui' is_json_ui %}
 
-{% include 'dhis2/partials/dhis2_dataset_map_templates.html' %}
+{% if is_json_ui %}
+  <div class="text-right">
+    <a href="?json=0">
+      <button type="button" class="btn btn-default">
+        {% trans "Switch to HTML interface" %}
+      </button>
+    </a>
+  </div>
+  {% include 'dhis2/partials/json_dhis2_dataset_map_template.html' %}
+
+{% else %}
+  <div class="text-right">
+    <a href="?json=1">
+      <button type="button" class="btn btn-default">
+        {% trans "Switch to JSON interface" %}
```

Comment:
> A JSON UI allows technical GS staff / MOTECH-trained partners to copy-and-paste config

From what I can tell, I believe that this is shown to everyone on a domain with access to this configuration page. Do you think that that is enough gate keeping?

Mostly asking this due to our history of UCR JSON interface where some domains were added and some partners were given access to the interface and shot themselves (or us) in the foot due to access to raw JSON allowing some very complex workflows or being able to mis-configure. Are there any similar concerns here? Is this significantly more powerful / easy to poorly configure than the non-JSON interface (as was the case with UCR)?

Reply:
This is a really good question.

Not that the HTML interface really has handrails, but it does prompt the user in a way that the JSON interface doesn't. You don't need the JSON UI in order to break integration, but it helps. :upside_down_face: 

Users can configure integration if their role has "Integration Access" permission ("Allow role to manage integration with other systems").

@dimagi/product would it make sense to disable this permission by default in the normal Admin role? That way it would have to be an intentional choice to enable it for users who have been vetted (not necessarily vetted by Dimagi, and not necessarily in a prescribed way, but at least chosen).